OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -395,6 +414,22 @@ public void onClick(View downloadView) {
         }
     }
 
+    private String getFormattedDueDate(final String date) throws IllegalArgumentException {
```

Comment:
Why are we throwing an `IllegalArgumentException` here ?

Reply:
Because `DateUtil.convertToDate(date)` & `SimpleDateFormat(String pattern)` can throw this exception in case if non-compatible due date format comes from server.